Source/NSStoryboard.m

```diff
@@ -0,0 +1,975 @@
+                  if ([[attr stringValue] isEqualToString: @"firstResponder"])
+                    {
+                      NSXMLNode *customClassAttr = [coel attributeForName: @"customClass"];
+                      NSXMLNode *idAttr = [coel attributeForName: @"id"];
+                      NSString *originalId = [idAttr stringValue];
+
+                      [idAttr setStringValue: @"-1"]; // set to first responder id
+                      [customClassAttr setStringValue: @"FirstResponder"];
+
+                      // Actions
+                      NSArray *cons = [document nodesForXPath: @"//action" error: NULL];
+                      NSEnumerator *consen = [cons objectEnumerator];
+                      NSXMLElement *celem = nil;
+
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *targetAttr = [celem attributeForName: @"target"];
+                          NSString *val = [targetAttr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [targetAttr setStringValue: @"-1"];
+                            }
+                        }
+
+                      // Outlets
+                      cons = [document nodesForXPath: @"//outlet" error: NULL];
+                      consen = [cons objectEnumerator];
+                      celem = nil;
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *attr = [celem attributeForName: @"destination"];
+                          NSString *val = [attr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [attr setStringValue: @"-1"];
+                            }
+                        }
+                    }
+                }
+
+              // Delete extra firstResponder...
+              /*
+              NSArray *ar = [document nodesForXPath: @"//customObject[@sceneMemberID=\"firstResponder\"]"
+                                              error: NULL];
+              if ([ar count] > 0)
+                {
+                  id node = [ar objectAtIndex: 0];
+                  [node detach];
+                }
+              */
+              
+              // Create document...
+              [_scenesMap setObject: document
+                             forKey: sceneId];
+              
+              // Map controllerId's to scenes...
+              if (controllerId != nil)
+                {
+                  [_controllerMap setObject: sceneId
+                                     forKey: controllerId];
+                }
+              
+              RELEASE(document);
+            }
+        }
+    }
+  else
+    {
+      NSLog(@"No document element in storyboard file");
+    }
+}
+
+- (NSMapTable *) _processSegues: (NSXMLDocument *)xmlIn
```

Comment:
Again a method that would benefit from being split up.

Reply:
Yes, indeed, both methods have gotten a bit big.   Again, I want to pull everything into another class and, possibly, do the XSLT change you suggested.

Reply:
While it could be done using XSLT, I believe doing it that way would defeat the idea of maintainability.  Also, I don't want to make XSLT a hard dependency for supporting this.   Additionally, I believe that doing it in ObjC is faster, though I have no tests to prove this.